### ISSUE TITLE
TheoryData with AutoFixture no longer works with xunit 2.9.2

### ISSUE THREAD

**Issue by dalyIsaac** [xunit/xunit#3034]
Starting in 2.9.2, I've been encountering the following error with `TheoryData`:

```log
System.InvalidOperationException : The test method expected 4 parameter values, but 2 parameter values were provided.
```

I think this error was likely introduced by 2.9.1, but masked by #3031 until 2.9.2.

The error occurs due to the fact that the `TheoryData` only contains 2 values, but I use AutoFixture to inject additional values via a custom attribute:

```csharp
public static TheoryData<string, int> GetData() => new()
{
		{ "this is the first string", 24 },
		{ "this is the second string with a much longer length", 51 }
};

[Theory, MemberAutoSubstituteData<ExampleCustomization>(nameof(GetData))]
public void TestWithAutoFixtureMember(string value, int length, Window window, Monitor monitor)
{
	Assert.Equal(value.Length, length);
	Assert.Equal("Main", window.Name);
	Assert.Equal(100, window.Height);
	Assert.Equal(200, window.Width);
	Assert.Equal(100, monitor.Height);
	Assert.Equal(200, monitor.Width);
}
```

The `MemberAutoSubstituteData` attribute extends the `MemberDataAttributeBase` class and injects a fixture based on the provided type. Equivalent attributes `AutoSubstituteData` and `InlineAutoSubstituteData` continue to work as expected.

This may also affect <https://github.com/Accenture/AutoFixture.XUnit2.AutoMock>, though I haven't tested that.

Repro project link: https://github.com/dalyIsaac/XunitAutoFixtureRepro

**Comment by bradwilson** [comment]
FWIW, `TheoryData<T1,T2>` was not changed in 2.9.1 (or 2.9.2). The fix in 2.9.2 was for `TheoryData` to implement a new internal interface so as to bypass any attempts to cast to `IEnumerable<object[]>`, as the covariance of the `T` in `IEnumerable<T>` is what was causing our issues.

I've never looked at the source to AutoFixture, so I don't know what it's doing and can't really offer any insight as to what is causing the new issue.

**Comment by dalyIsaac** [comment]
That makes sense. I was relying on `ConvertDataItem` being called, but the new specific handling of `ITheoryData` means `ConvertDataItem` is never called.

Thanks!
